lib/rspec/core/reporter.rb

```diff
@@ -53,6 +55,13 @@ def register_listener(listener, *notifications)
     end
 
     # @private
+    def prepare_default(loader, output_stream, deprecation_stream)
+      @setup_default = lambda do
+        loader.setup_default output_stream, deprecation_stream
```

Comment:
tbh this all feels a bit janky and complicated but I don't have any better ideas :(